File: ASTC-Simulation/src/Point.java
public class Point{
    private int x;
    private int y;

    public Point(){
        this.x = 0;
        this.y = 0;
    }

    public Point(int x, int y){
        this.x = x;
        this.y = y;
    }

    public void setX(int x){
        this.x = x;
    }

    public int getX(){
        return this.x;
    }

    public void setY(int y){
        this.y = y;
    }

    public int getY(){
        return this.y;
    }

    public void setPoint(int x, int y){
        this.x = x;
        this.y = y;
    }

    public void setPoint(Point p){
        this.x = p.getX();
        this.y = p.getY();
    }

    public Point getPoint(){
        return this;
    }

    public void move(int dx, int dy){
        this.x += dx;
        this.y += dy;
    }

    public void move(Point p){
        this.x += p.getX();
        this.y += p.getY();
    }

    public void moveTo(int x, int y){
        this.x = x;
        this.y = y;
    }

    public void moveTo(Point p){
        this.x = p.getX();
        this.y = p.getY();
    }

    public double distanceTo(int x, int y){
        return Math.sqrt(Math.pow(this.x - x, 2) + Math.pow(this.y - y, 2));
    }

    public double distanceTo(Point p){
        return Math.sqrt(Math.pow(this.x - p.getX(), 2) + Math.pow(this.y - p.getY(), 2));
    }

    public String toString(){
        return "(" + this.x + ", " + this.y + ")";
    }

    public boolean equals(Point p){
        return this.x == p.getX() && this.y == p.getY();
    }

    public boolean equals(int x, int y){
        return this.x == x && this.y == y;
    }

    public boolean equals(Object o){
        if(o instanceof Point){
            Point p = (Point) o;
            return this.x == p.getX() && this.y == p.getY();
        }
        return false;
    }

    public Point clone(){
        return new Point(this.x, this.y);
    }
}

File: Lab3/point.cpp
#include <iostream>
#include <math.h>
using namespace std;

class Point{
    float x, y;

    public:


    Point(float x1 = 0, float y1 = 0){
        x = x1;
        y = y1;
    }


    Point (Point & P){
        x = P.x;
        y = P.y;
    }

    //setters

    void setX(float x){
        x = x;
    }

    void setY(float y){
        y = y;
    }

    //Getters

    float getX(){
        return x;
    }

    float getY(){
        return y;
    }

    //Input fucntion

    Point input(){
        cout << "Point x: ";
        cin >> x;
        cout << "Point y: ";
        cin >> y;
        return Point(x, y);
    }

    float distance(Point P)
    {
        return (sqrt(pow((x - P.x), 2) + pow((y - P.y),2)));
    }

    bool isZero(){
        if ((x == 0) && (y == 0))
            return true;
       return false;
    }

    Point MidPoint(){
        return Point((x/2), (y/2));
    }

    bool isEqualTo(Point &P){
        if((x == P.x) && (y == P.y))
            return true;
        return false;
    }

    bool isGreater(Point &P){
        float d1 = distance(Point(0, 0));
        float d2 = P.distance(Point(0, 0));

        if (d1 > d2){
            return true;
        }

        return false;
    }

    void output(){
        cout << "X: " << x << " Y: " << y << endl;
    }

};


int main(){
    Point p1;
    p1.input();
    p1.output();
    Point p2(3, 4);
    p2.output();
    cout << "Distance between P1 and p2 is : " << p1.distance(p2) << endl;
    cout << "Is p1 greater than p2: " << p1.isGreater(p2) << endl;
    Point p3 = p1;
    cout <<"IS P3 equals to p2: " <<  p3.isEqualTo(p2) << endl;
    p2.output();
    Point p4 = p2.MidPoint();
    cout << "Midpoints of p2 are: ";
    p4.output();

}